src/main/java/org/jenkinsci/plugins/pipeline/utility/steps/zip/ZipStepExecution.java

```diff
@@ -120,23 +120,30 @@ protected Void run() throws Exception {
     static class ZipItFileCallable extends MasterToSlaveFileCallable<Integer> {
         final FilePath zipFile;
         final String glob;
+        final boolean overwrite;
 
-        public ZipItFileCallable(FilePath zipFile, String glob) {
+        public ZipItFileCallable(FilePath zipFile, String glob, boolean overwrite) {
             this.zipFile = zipFile;
             this.glob = StringUtils.isBlank(glob) ? "**/*" : glob;
+            this.overwrite = overwrite;
         }
 
         @Override
         public Integer invoke(File dir, VirtualChannel channel) throws IOException, InterruptedException {
-            String canonicalZip = new File(zipFile.getRemote()).getCanonicalPath();
+            File zip = new File(zipFile.getRemote());
+            if (overwrite && zip.exists()) {
+                if (!zip.delete()) {
```

Comment:
https://docs.oracle.com/javase/8/docs/api/java/nio/file/Files.html#delete-java.nio.file.Path- would be a better File management API

Reply:
I changed it to the api you suggested. 
@oleg-nenashev can you check if it is as expected?
Thank you very much